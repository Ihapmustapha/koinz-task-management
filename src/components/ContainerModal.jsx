import React from "react";
import PropTypes from "prop-types";
// components
import {
  Dialog,
  DialogActions,
  DialogTitle,
  DialogContent,
  DialogContentText,
  Button,
} from "@material-ui/core";
// redux
import { connect } from "react-redux";
import * as actions from "../store/actions";

const ContainerDialog = ({
  children,
  isOpen,
  title,
  description,
  secondaryButtonText,
  primaryButtonAction,
  primaryButtonText,
  closeModal,
}) => {
  const handleClose = () => {
    closeModal();
  };
  return (
    <>
      <Dialog
        open={isOpen}
        onClose={handleClose}
        aria-labelledby="dialog-title"
        aria-describedby="dialog-description"
      >
        {title && <DialogTitle id="dialog-title">{title}</DialogTitle>}
        <DialogContent>
          {description && (
            <DialogContentText id="dialog-description">
              {description}
            </DialogContentText>
          )}
          {children}
        </DialogContent>
        <DialogActions>
          <Button color="secondary" onClick={handleClose}>
            {secondaryButtonText}
          </Button>
          <Button color="primary" onClick={primaryButtonAction} autoFocus>
            {primaryButtonText}
          </Button>
        </DialogActions>
      </Dialog>
    </>
  );
};

ContainerDialog.defaultProps = {
  children: null,
  title: null,
  description: null,
};

ContainerDialog.propTypes = {
  children: PropTypes.node,
  isOpen: PropTypes.bool.isRequired,
  title: PropTypes.string,
  description: PropTypes.string,
  closeModal: PropTypes.func.isRequired,
  secondaryButtonText: PropTypes.string.isRequired,
  primaryButtonAction: PropTypes.func.isRequired,
  primaryButtonText: PropTypes.string.isRequired,
};

const mapStateToProps = (state) => ({
  isOpen: state?.modalState?.isOpen || false,
});
const mapDispatchToProps = (dispatch) => ({
  closeModal: () => dispatch(actions.closeModal()),
});

export default connect(mapStateToProps, mapDispatchToProps)(ContainerDialog);
