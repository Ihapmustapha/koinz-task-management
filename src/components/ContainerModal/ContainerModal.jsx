import React from "react";
import PropTypes from "prop-types";
// components
import {
  Dialog,
  DialogTitle,
  DialogContent,
  DialogContentText,
} from "@material-ui/core";
// redux
import { connect } from "react-redux";
import * as actions from "../../store/actions";

const ContainerModal = ({
  children,
  isOpen,
  title,
  description,
  closeModal,
}) => {
  const handleClose = () => {
    closeModal();
  };
  return (
    <>
      <Dialog
        open={isOpen}
        onClose={handleClose}
        aria-labelledby="dialog-title"
        aria-describedby="dialog-description"
        keepMounted={false}
      >
        {title && <DialogTitle id="dialog-title">{title}</DialogTitle>}
        <DialogContent>
          {description && (
            <DialogContentText id="dialog-description">
              {description}
            </DialogContentText>
          )}
          {children}
        </DialogContent>
      </Dialog>
    </>
  );
};

ContainerModal.defaultProps = {
  children: null,
  title: null,
  description: null,
};

ContainerModal.propTypes = {
  children: PropTypes.node,
  isOpen: PropTypes.bool.isRequired,
  title: PropTypes.string,
  description: PropTypes.string,
  closeModal: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  isOpen: state?.modalState?.isOpen || false,
});
const mapDispatchToProps = (dispatch) => ({
  closeModal: () => dispatch(actions.closeModal()),
});

export default connect(mapStateToProps, mapDispatchToProps)(ContainerModal);
