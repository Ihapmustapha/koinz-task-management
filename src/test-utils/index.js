export { default as MockedProvider } from "./mockedProvider";
