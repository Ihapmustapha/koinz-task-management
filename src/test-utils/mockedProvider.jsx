import React from "react";
import PropTypes from "prop-types";
import { Provider } from "react-redux";
import { configureStore } from "../store";

const MockedProvider = ({ initialState, children }) => (
  <Provider store={configureStore(initialState)}>{children}</Provider>
);

MockedProvider.propTypes = {
  initialState: PropTypes.shape({}).isRequired,
  children: PropTypes.arrayOf(PropTypes.node).isRequired,
};

export default MockedProvider;
