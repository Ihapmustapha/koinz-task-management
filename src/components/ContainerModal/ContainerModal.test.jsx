import React from "react";
import { mount } from "enzyme";
import {
  DialogTitle,
  DialogContentText,
  DialogContent,
} from "@material-ui/core";
import ContainerModal from "./ContainerModal";
import { MockedProvider } from "../../test-utils";

const getWrapper = (props) =>
  mount(
    <MockedProvider initialState={{ modalState: { isOpen: true } }}>
      <ContainerModal {...props} />
    </MockedProvider>
  );

let wrapper = getWrapper({});

describe("<ContainerModal />", () => {
  it("should match the snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render a title", () => {
    const title = "dummy title";

    wrapper = getWrapper({ title });

    expect(wrapper.find(DialogTitle).exists()).toBe(true);
    expect(wrapper.find(DialogTitle).text()).toBe(title);
  });

  it("should render a description", () => {
    const description = "dummy description";

    wrapper = getWrapper({ description });

    expect(wrapper.find(DialogContentText).exists()).toBe(true);
    expect(wrapper.find(DialogContentText).text()).toBe(description);
  });

  it("should render the children inside the Content", () => {
    const DummyContent = () => <div>Dummy Content</div>;

    wrapper = getWrapper({ children: <DummyContent /> });

    expect(wrapper.find(DialogContent).find(DummyContent).exists()).toBe(true);
  });
});
