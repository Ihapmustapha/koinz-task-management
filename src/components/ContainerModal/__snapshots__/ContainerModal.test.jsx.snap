// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContainerModal /> should match the snapshot 1`] = `"<div role=\\"presentation\\" class=\\"MuiDialog-root\\" style=\\"position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;\\"><div class=\\"MuiBackdrop-root\\" aria-hidden=\\"true\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div><div tabindex=\\"0\\" data-test=\\"sentinelStart\\"></div><div class=\\"MuiDialog-container MuiDialog-scrollPaper\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\" role=\\"none presentation\\" tabindex=\\"-1\\"><div class=\\"MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-elevation24 MuiPaper-rounded\\" role=\\"dialog\\" aria-describedby=\\"dialog-description\\" aria-labelledby=\\"dialog-title\\"><div class=\\"MuiDialogContent-root\\"></div></div></div><div tabindex=\\"0\\" data-test=\\"sentinelEnd\\"></div></div>"`;
